### PR TITLE
Snapshot portability across libvirt 3 and 6

### PR DESCRIPTION
**Pull request by sam-github**
There have been changes to how libvirt creates snapshots, that make the
current go-qemu APIs unable to work on libvirt 6, since they relied
on using a device string that is no longer present.

Fortunately, the call sequence and arguments that work for libvirt 6
*also* work for libvirt 3.

Unfortunately, they require information specifically about the block job
being cancelled, so the API changes slightly.

Block jobs were being completed/cancelled on the BlockDevice,
but the information used to identify them in libvirt6 is not the
BlockDevice.Device, but the BlockJob.Device, so move the methods to the
BlockJob.

See:
- https://www.mail-archive.com/qemu-discuss@nongnu.org/msg05225.html

### CODE REVIEW

File: qemu/block.go [hunk 1]
```diff
@@ -189,15 +190,15 @@ func (bd BlockDevice) CancelJob(d *Domain, timeout time.Duration) error {
 	})
 }
 
-// CompleteJob finalizes any running block jobs on the provided block device.
+// CompleteJob finalizes a running block job.
```
Comment (benlemasurier) [inline qemu/block.go:193]: `Complete`

File: qemu/block.go [hunk 2]
```diff
@@ -173,14 +174,14 @@ func (bd BlockDevice) Commit(d *Domain, overlay string, timeout time.Duration) e
 	})
 }
 
-// CancelJob cancels any active block jobs on a block device.
+// CancelJob cancels a active block job.
```
Comment (benlemasurier) [inline qemu/block.go:177]: // _Cancel_ cancels _an_ active block job.

**Review by benlemasurier — COMMENTED** [review]
two trivial comments to fix the API docs, otherwise lgtm. If travis ran I think `golint` would be complaining.

File: qemu/block.go (hunk 2 above)
Comment (sam-github) [inline qemu/block.go:177]:
```suggestion
// CancelJob cancels an active block job.
```

File: qemu/block.go (hunk 1 above)
Comment (sam-github) [inline qemu/block.go:193]:
```suggestion
// Complete finalizes a running block job.
```

File: qemu/block.go (hunk 2 above)
Comment (benlemasurier) [reply qemu/block.go:177]: I care less about the `a` `an` thing, more about the function name (it was renamed to `Cancel()`, no?

**Comment by sam-github** [comment]
I ran the travis checks locally. I assume travis isn't building because I'm not a collaborator.

@benlemasurier can you kick travis, or give me perms to do so?

**Review by benlemasurier — APPROVED** [review]

**Comment by benlemasurier** [comment]
> I ran the travis checks locally. I assume travis isn't building because I'm not a collaborator.
> 
> @benlemasurier can you kick travis, or give me perms to do so?

I think travis-ci is basically dead at this point unless we start paying them. I've got a branch to switch over to GH actions but it still needs some work.

I'll test locally and force the merge if I can.

**Comment by sam-github** [comment]
Ah, ok. I was about to PR some updates to the test matrix, I won't bother! :-)

**Pull request merged**